python/vowpalwabbit/pyvw.py

```diff
@@ -1857,243 +1929,410 @@ def process(self, df=None):
 
 class DFtoVW:
     """Convert a pandas DataFrame to a suitable VW format.
-    Instances of this class are built with classes such as SimpleLabel, Feature
-    or Namespace (that can themselves be built on Col object(s)).
+    Instances of this class are built with classes such as SimpleLabel,
+    Feature or Namespace.
+
     The class also provided a convenience constructor to initialize the class
     based on the target/features column names only.
     """
 
-    def __init__(self, df, namespaces, label=None, tag=None):
+    def __init__(
+        self,
+        df,
+        features=None,
```

Comment:
what happens when we set both the features and the namespaces, that would always imply an error?

Reply:
or is there a valid scenario when we would use the init and set both features and namespaces when calling into this function?

Reply:
Error will always be raised if both `features` and `namespaces` are set. 
@lalo 

I see two distinct cases: 
- `features` is used to create naked namespace, 
- `namespaces` is used to create a single named/scaled namespace or multiples named/scaled namespaces.

Let's imagine an user that want to build a naked namespace with `features` and 2 named namespaces with `namespaces`. This operation could be performed using just the `namespaces` parameter, by passing a list of 3 namespaces to it (the first element of the list being the unnamed namespace for example).

 I see `features` as shortcut/convenience parameter for new/rushed users, and `namespaces` for more advanced usages. But allowing both parameters to be passed might be a bit confusing (especially regarding the location where the naked namespace should be). What do you think ?

Reply:
Yes, your last point was what I was going with but at least we error out right now if we supply both. I feel it is just a matter of coming up with better naming and get feedback from more people. It just might be misleading to have both parameters. For now, maybe just add on the comment of this function that it only takes one or the other. Later we can shift things around more.